pr_agent/algo/utils.py

```diff
@@ -1,5 +1,8 @@
 from __future__ import annotations
 
+import json
+import logging
+import re
 import textwrap
 
```

Comment:
Consider adding a default value for `max_iter` in the function signature. This will make the function more flexible and prevent infinite loops if the function is called without a `max_iter` argument. [important]